### PR TITLE
Fixed wrong crit from FFLogs and also some other minor fixes

### PR DESCRIPTION
**Pull request by jplippi**
- fixed wrong crit info from dots from FFLogs
- fixed arcanum not considering 'refreshbuff'
- fixed webpackChunkName :blobknife:

### CODE REVIEW

File: src/parser/jobs/brd/modules/AdditionalStats.js [hunk 1]
```diff
@@ -100,25 +102,55 @@ export default class AdditionalStats extends Module {
...
+					player.statuses[event.ability.guid] = {
+						isActive: event.type.startsWith('apply') || event.type.startsWith('refresh'),
+					}
+
+					// If the current status is a crit modifier, we add the strength onto it to use later
+					const critModifier = CRIT_MODIFIERS.find(cm => cm.id === event.ability.guid)
+					if (critModifier) {
+
+						// If it's a spear card, we get the modifier from the event, thank's to arcanum
+						if (critModifier.id === STATUSES.THE_SPEAR.id) {
```
Comment (ackwell) [inline src/parser/jobs/brd/modules/AdditionalStats.js]: nit; you're using `critModifier.strength` in both cases, can move the base modifier out of the conditional and just have the conditional case for spear funtimes.

File: src/parser/jobs/brd/modules/AdditionalStats.js [hunk 2]
```diff
@@ -100,25 +102,55 @@ export default class AdditionalStats extends Module {
...
-					enemy.statuses[event.ability.guid] = event.type.startsWith('apply') || event.type.startsWith('refresh')
+					enemy.statuses[event.ability.guid] = {
+						isActive: event.type.startsWith('apply') || event.type.startsWith('refresh'),
+					}
+					// If the current status is a crit modifier, we add the strength onto it to use later
+					const critModifier = CRIT_MODIFIERS.find(cm => cm.id === event.ability.guid)
+					if (critModifier) {
+
+						// If it's a spear card, we get the modifier from the event, thank's to arcanum
+						if (critModifier.id === STATUSES.THE_SPEAR.id) {
```
Comment (ackwell) [inline src/parser/jobs/brd/modules/AdditionalStats.js]: Did you forget to get the strength?

File: src/parser/jobs/brd/modules/AdditionalStats.js [hunk 3]
```diff
@@ -100,25 +102,55 @@ export default class AdditionalStats extends Module {
...
 				// In case it's a status on an enemy, we attribute it to that specific enemy by the ID
 				} else if (!event.targetIsFriendly) {
 					const enemy = this._getEnemy(event.targetID)
-					enemy.statuses[event.ability.guid] = event.type.startsWith('apply') || event.type.startsWith('refresh')
+					enemy.statuses[event.ability.guid] = {
+						isActive: event.type.startsWith('apply') || event.type.startsWith('refresh'),
+					}
+					// If the current status is a crit modifier, we add the strength onto it to use later
+					const critModifier = CRIT_MODIFIERS.find(cm => cm.id === event.ability.guid)
+					if (critModifier) {
```
Comment (ackwell) [inline src/parser/jobs/brd/modules/AdditionalStats.js]: Would be good to DRY this logic up a bit.

File: src/parser/jobs/brd/modules/AdditionalStats.js [hunk 4]
```diff
@@ -174,6 +206,18 @@ export default class AdditionalStats extends Module {
...
+					// First of all, let's fix cases of broken crit
+					event.expectedCritRate -= accumulatedCritBuffs * 1000
+
+					while (event.expectedCritRate < BASE_CRIT_PROBABILITY) {
+						event.expectedCritRate += 256
+					}
+
+					event.expectedCritRate += accumulatedCritBuffs * 1000
+
+					console.log(event.expectedCritRate)
```
Comment (ackwell) [inline src/parser/jobs/brd/modules/AdditionalStats.js]: loggyboi

File: src/parser/jobs/brd/modules/AdditionalStats.js [hunk 5]
```diff
@@ -272,15 +316,13 @@ export default class AdditionalStats extends Module {
 
 		for (const modifier of CRIT_MODIFIERS) {
 
-			let hasStatus = false
+			const enemyStatus = this._hasStatus(enemy, modifier.id)
```
Comment (ackwell) [inline src/parser/jobs/brd/modules/AdditionalStats.js]: `_hasStatus` probably isn't the best name for something that _gets_ a status.

File: src/parser/jobs/brd/modules/AdditionalStats.js [hunk 6]
```diff
@@ -387,6 +429,11 @@ export default class AdditionalStats extends Module {
 				}
 			}
 		}
+		// If there are no damage instances, we can't really calculate K. Let's trigger brokenLog and return 1
+		if (!values || !values.length) {
+			this.brokenLog.trigger()
```
Comment (ackwell) [inline src/parser/jobs/brd/modules/AdditionalStats.js]: Wary of triggering `brokenLog` here - I mean, no damage in a log is _weird_, but not necessarily _broken_. I think returning a sane baseline K is reasonable enough.

File: src/parser/jobs/brd/modules/AdditionalStats.js (hunk 2 above)
Comment (jplippi) [reply src/parser/jobs/brd/modules/AdditionalStats.js]: Did I? I'm confused. I'm gonna set as resolved and push my changes. Let me know if it's still there.

File: src/parser/jobs/brd/modules/AdditionalStats.js [hunk 7]
```diff
@@ -147,18 +164,18 @@ export default class AdditionalStats extends Module {
...
-						if (this._hasStatus(this._getEnemy(event.targetID), STATUSES.PIERCING_RESISTANCE_DOWN.id)) {
+						if (this._getStatus(this._getEnemy(event.targetID), STATUSES.PIERCING_RESISTANCE_DOWN.id)) {
 							fixedMultiplier = Math.trunc((fixedMultiplier + DISEMBOWEL_STRENGTH) * 100) / 100
 						}
 						// AND ALSO FOR RANGED TRAIT, BECAUSE APPARENTLY IT'S PHYSICAL DAMAGE ONLY REEEEEEEEEE
 						fixedMultiplier = Math.trunc((fixedMultiplier + TRAIT_STRENGTH) * 100) / 100
 					}
 
-					// Collects the damage instances, to be used for calculating crit and 'K'
+					// Collects the damage instances, to be used for calculating crit and 'potencyDamageRatio'
```
Comment (ackwell) [inline src/parser/jobs/brd/modules/AdditionalStats.js]: :bloblove:

**Pull request merged**